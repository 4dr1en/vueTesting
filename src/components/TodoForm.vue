<script setup lang="ts">
import { ref } from "vue";
const todo = ref<string>("");

const emit = defineEmits<{
  (e: "addTodo", todo: string): void;
}>();

function sendTodo() {
  if (!todo.value) return;
  emit("addTodo", todo.value);
  todo.value = "";
}
</script>

<template>
  <div>
    <label for="todo">Todo</label>
    <input type="text" v-model="todo" @keydown.enter="sendTodo" id="todo" />
    <button @click="sendTodo">Add</button>
  </div>
</template>

<style lang="scss"></style>
