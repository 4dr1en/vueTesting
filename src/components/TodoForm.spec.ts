import { mount } from "@vue/test-utils";
import TodoForm from "@/components/TodoForm.vue";
import { expect, describe, it } from "vitest";

describe("TodoForm",  () => {
	it("The field is reset after submit", async() => {
		const wrapper = mount(TodoForm);
		const input = wrapper.find("input");
		await input.setValue("Buy groceries");
		const button = wrapper.find("button");
		await button.trigger("click");

		expect(input.text()).toBe("");
	});

	it("Emits a 'addTodo' event when the form is submitted", async() => {
		const wrapper = mount(TodoForm);
		const input = wrapper.find("input");
		await input.setValue("Buy groceries");
		const button = wrapper.find("button");
		await button.trigger("click");

		expect(wrapper.emitted("addTodo")).toHaveLength(1);
		expect(wrapper.emitted("addTodo")![0]![0]).toBe("Buy groceries");
	});
});