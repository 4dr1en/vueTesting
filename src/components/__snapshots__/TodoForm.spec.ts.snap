// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoForm > Should match the snapshot 1`] = `"<div><label for=\\"todo\\">Todo</label><input type=\\"text\\" id=\\"todo\\"><button>Add</button></div>"`;
